lightningd/plugin.c

```diff
@@ -1385,6 +1385,59 @@ static struct command_result *plugin_rpcmethod_dispatch(struct command *cmd,
 	return command_still_pending(cmd);
 }
 
+static const char *plugin_rpcmethod_clnrest_add(struct plugin *plugin, 
+                                                const char *buffer, 
+                                                const jsmntok_t *tok, 
+                                                struct json_command *cmd) 
+{
+	const jsmntok_t *clnresttok = json_get_member(buffer, tok, "clnrest");
+
+	if (!clnresttok) {
+			cmd->clnrest = NULL;
+			return NULL;
+	}
+
+	struct clnrest *clnrest = tal(cmd, struct clnrest);
+
+	// TODO: check each field for validity
+
+	const jsmntok_t *pathtok = json_get_member(buffer, clnresttok, "path");
+	const jsmntok_t *methodtok = json_get_member(buffer, clnresttok, "method");
+	const jsmntok_t *contenttypetok = json_get_member(buffer, clnresttok, "content_type");
+	const jsmntok_t *runetok = json_get_member(buffer, clnresttok, "rune");
+	const jsmntok_t *pluginnametok = json_get_member(buffer, tok, "name");
+
+	if (pathtok) {
+		clnrest->path = json_strdup(clnrest, buffer, pathtok);
+	} else {
+		clnrest->path = json_strdup(clnrest, buffer, pluginnametok);
+	}
+
+	if (methodtok) {
+		clnrest->method = json_strdup(clnrest, buffer, methodtok);
+	} else {
+		clnrest->method = "GET";
```

Comment:
Default should be POST if we decide to keep it.

Reply:
updated default value to POST in [d52a311](https://github.com/ElementsProject/lightning/pull/7507/commits/d52a3114734a862343282d67d0fde5af5e4274b6)